app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -420,9 +420,10 @@ BrowserWorld::State::UpdateControllers(bool& aRelayoutWidgets) {
       controller.pointer->Load(device);
     }
 
-    if ((!(controller.lastButtonState & ControllerDelegate::BUTTON_APP) && (controller.buttonState & ControllerDelegate::BUTTON_APP)) ||
-      (!(controller.lastButtonState & ControllerDelegate::BUTTON_B) && (controller.buttonState & ControllerDelegate::BUTTON_B)) ||
-      (!(controller.lastButtonState & ControllerDelegate::BUTTON_Y) && (controller.buttonState & ControllerDelegate::BUTTON_Y))) {
+    if (controller.mode == ControllerMode::Device &&
```

Comment:
Why this check for the ControllerMode is only needed for the simulateBack logic ? Have you checked whether there are similar problems with the focus / click events, controlled now with the X / A buttons ?

Reply:
I also think that the fix is correct, but perhaps we should aim for a more general solution. I've indeed seen many of these issues, even with only using hand tracking

Reply:
@javifernandez @svillar When you pick the controller, `UpdateControllers()` is called with the following values:

- mode = `ControllerMode::Hand`
- `controller.lastButtonState & ControllerDelegate::BUTTON_APP` is false (0)
- `controller.buttonState & ControllerDelegate::BUTTON_APP` is true (not 0)

This gets interpreted as a click. Checking that we are in controller mode fixes this problem and I think that we should merge this change.